src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -189,7 +192,9 @@ TryToDelegateFunctionCall(Query *query)
 		return NULL;
 	}
 
-	if (expression_tree_walker((Node *) funcExpr->args, contain_param_walker, NULL))
+	*hasParam = expression_tree_walker((Node *) funcExpr->args, contain_param_walker,
```

Comment:
I think this should get the param kind so it can differentiate between `PARAM_EXTERN`, `PARAM_SUBLINK`, ...

The returned hasParam also needs to be renamed to `hasExternParam` (if my understanding is correct that prepared statements use `PARAM_EXTERN`. Please double check)

Reply:
Updated